gcloud/datastore/query.py

```diff
@@ -140,441 +243,266 @@ def filter(self, property_name, operator, value):
+    @property
+    def group_by(self):
+        """Names of fields used to group query results.
 
-        :type limit: integer
-        :param limit: An optional limit to apply temporarily to this query.
-                      That is, the Query itself won't be altered,
-                      but the limit will be applied to the query
-                      before it is executed.
-
-        :rtype: tuple of mixed types
-        :returns: The first entry is a :class:`gcloud.datastore.entity.Entity`
-                  list matching this query's criteria. The second is a base64
-                  encoded cursor for paging and the third is a boolean
-                  indicating if there are more results.
-        :raises: `ValueError` if more_results is not one of the enums
-                 NOT_FINISHED, MORE_RESULTS_AFTER_LIMIT, NO_MORE_RESULTS.
+        :rtype: sequence of string
         """
-        clone = self
-
-        if limit:
-            clone = self.limit(limit)
-
-        query_results = self.dataset().connection().run_query(
-            query_pb=clone.to_protobuf(),
-            dataset_id=self.dataset().id(),
-            namespace=self._namespace,
-            )
-        # NOTE: `query_results` contains an extra value that we don't use,
-        #       namely `skipped_results`.
-        #
-        # NOTE: The value of `more_results` is not currently useful because
-        #       the back-end always returns an enum
-        #       value of MORE_RESULTS_AFTER_LIMIT even if there are no more
-        #       results. See
-        #       https://github.com/GoogleCloudPlatform/gcloud-python/issues/280
-        #       for discussion.
-        entity_pbs, cursor_as_bytes, more_results_enum = query_results[:3]
-
-        entities = [helpers.entity_from_protobuf(entity,
-                                                 dataset=self.dataset())
-                    for entity in entity_pbs]
+        return self._group_by[:]
 
-        cursor = base64.b64encode(cursor_as_bytes)
+    @group_by.setter
+    def group_by(self, value):
+        """Set fields used to group query results.
 
-        if more_results_enum == self._NOT_FINISHED:
-            more_results = True
-        elif more_results_enum in self._FINISHED:
-            more_results = False
-        else:
-            raise ValueError('Unexpected value returned for `more_results`.')
-
-        return entities, cursor, more_results
-
-    def fetch(self, limit=None):
-        """Executes the Query and returns matching entities
+        :type value: string or sequence of strings
+        :param value: Each value is a string giving the name of a
+                         property to use to group results together.
+        """
+        if isinstance(value, str):
+            value = [value]
+        self._group_by[:] = value
 
-        This calls `fetch_page()` but does not use the paging information.
+    def fetch(self, limit=0, offset=0, start_cursor=None, end_cursor=None):
```

Comment:
Do we know if `0` causes different backend behavior than unset for `limit` and `offset`?